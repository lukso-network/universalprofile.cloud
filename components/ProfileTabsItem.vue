<script setup lang="ts">
import { tv } from 'tailwind-variants'

type Props = {
  label: string
  isActive: boolean
  count?: number
}

const props = defineProps<Props>()

const styleVariants = tv({
  slots: {
    tab: 'heading-inter-17-semi-bold flex items-center',
    counter:
      'paragraph-inter-10-semi-bold ml-2 rounded-4 border border-neutral-20 px-1 py-px',
  },
  variants: {
    isActive: {
      true: {
        tab: 'cursor-default',
        counter: 'bg-neutral-20 text-neutral-100',
      },
      false: {
        tab: 'cursor-pointer opacity-50 transition hover:opacity-100',
        counter: 'bg-transparent text-neutral-20',
      },
    },
  },
})

const styles = computed(() => {
  return styleVariants({
    isActive: props.isActive,
  })
})
</script>

<template>
  <li :class="styles.tab()">
    {{ label }}
    <span v-if="count" :class="styles.counter()">{{ count }}</span>
  </li>
</template>
