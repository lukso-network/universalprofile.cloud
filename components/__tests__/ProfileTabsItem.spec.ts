import { renderSuspended } from '@nuxt/test-utils/runtime'
import { describe, expect, it } from 'vitest'

import ProfileTabsItem from '../ProfileTabsItem.vue'

describe('ProfileTabsItem', () => {
  it('renders default state', async () => {
    const component = await renderSuspended(ProfileTabsItem, {
      props: {
        label: 'Test Label',
        isActive: false,
        count: 0,
      },
    })
    expect(component.html()).toMatchSnapshot()
  })

  it('renders active state', async () => {
    const component = await renderSuspended(ProfileTabsItem, {
      props: {
        label: 'Test Label',
        isActive: true,
        count: 0,
      },
    })
    expect(component.html()).toMatchSnapshot()
  })

  it('renders count', async () => {
    const component = await renderSuspended(ProfileTabsItem, {
      props: {
        label: 'Test Label',
        isActive: false,
        count: 2,
      },
    })
    expect(component.html()).toMatchSnapshot()
  })
})
