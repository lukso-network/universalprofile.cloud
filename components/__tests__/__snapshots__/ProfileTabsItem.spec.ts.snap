// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProfileTabsItem > renders active state 1`] = `
"<div id="test-wrapper">
  <li class="heading-inter-17-semi-bold flex items-center cursor-default">Test Label
    <!--v-if-->
  </li>
</div>"
`;

exports[`ProfileTabsItem > renders count 1`] = `
"<div id="test-wrapper">
  <li class="heading-inter-17-semi-bold flex items-center cursor-pointer opacity-50 transition hover:opacity-100">Test Label <span class="paragraph-inter-10-semi-bold ml-2 rounded-4 border border-neutral-20 px-1 py-px bg-transparent text-neutral-20">2</span></li>
</div>"
`;

exports[`ProfileTabsItem > renders default state 1`] = `
"<div id="test-wrapper">
  <li class="heading-inter-17-semi-bold flex items-center cursor-pointer opacity-50 transition hover:opacity-100">Test Label
    <!--v-if-->
  </li>
</div>"
`;
